src/host/srvinit.cpp

```diff
@@ -823,19 +817,25 @@ PWSTR TranslateConsoleTitle(_In_ PCWSTR pwszConsoleTitle, const BOOL fUnexpand,
 
         THROW_IF_FAILED(localPointerToThread->Initialize(g.pRender));
 
-        // Allow the renderer to paint.
-        g.pRender->EnablePainting();
-
         // Set up the renderer to be used to calculate the width of a glyph,
         //      should we be unable to figure out its width another way.
         auto pfn = std::bind(&Renderer::IsGlyphWideByFont, static_cast<Renderer*>(g.pRender), std::placeholders::_1);
         SetGlyphWidthFallback(pfn);
     }
     catch (...)
     {
-        Status = NTSTATUS_FROM_HRESULT(wil::ResultFromCaughtException());
+        return NTSTATUS_FROM_HRESULT(wil::ResultFromCaughtException());
     }
 
+    NTSTATUS Status = SetUpConsole(&p->ConsoleInfo, p->TitleLength, p->Title, p->CurDir, p->AppName);

```

Comment:
Given that you cannot see how `g.pRender` is a dependency of `SetUpConsole` now... we should perhaps leave a comment specifying that this has to be the order of them because setting up the console will go look it up and it shouldn't be reordered carelessly.

Reply:
Yeah, that's a good point. I've added some comments here now.